### ISSUE TITLE
Consider button focus rings on immediate child

### ISSUE THREAD

**Issue by danrabbit** [elementary/stylesheet#1056]
## Problem

We have some issues with focus rings being outside of buttons like clipping in linked styles or when there isn’t technically any margin around the button or some issue with z axis ordering or clipping in overlays. Lots of little edge cases that make focus rings outside of the button break

## Proposal

Maybe instead, we should apply the focus style to the immediate child of the button (likely a label or icon) as a background 

## Prior Art

This would look more similar to the flat button focus styles and indeed maybe we could combine a lot of the focus style code this way as well

**Comment by danrabbit** [comment]
In practice, looks like:

![Screenshot from 2021-06-30 12 47 31](https://user-images.githubusercontent.com/7277719/124022247-69079a00-d9a1-11eb-93af-7e228ab7dd29.png)
![Screenshot from 2021-06-30 12 46 42](https://user-images.githubusercontent.com/7277719/124022248-69a03080-d9a1-11eb-819f-870bd1b7ca08.png)

**Comment by cassidyjames** [comment]
Hmmmmm... functionally I think it makes sense, but it's quite an aesthetic departure. Have you investigated using inset shadows and/or outline with a negative `outline-offset` to keep the outlined look but placing the glow on the inside of the widget instead of the outside?

**Comment by danrabbit** [comment]
Yeah, I'm not in love with it, but I wanted to record it somewhere as a possible solution.

I haven't spend any time working on a solution for outlines inside of elements
